### DISCUSSION TITLE
Confusion on the instruction project 3

### DISCUSSION THREAD

**Discussion by BakariMtili** [IBAHCM/RPiR#344] [Archive] [answered]
Hello, @richardreeve and @soniamitchell I am slightly confused with some bits of instructions from project 3. If I can recall correctly the instructions on some of the specific functions such as Berger parker has some additional information from the general diversity function (which means that we won't get the same results by using the general function and some specific function for a specific value of q). At the end of the instructions, we are required to create a demo by calling the general function rather than the specific functions with the additional information.
Therefore my questions are: why did we create the specific functions? Is using the general function in the demo supposed to bring the same output as the specific function with a specific q value?
Please some more clarification will be super helpful.
Thank you so much for your time

**Comment by richardreeve** [comment, marked as answer]
The specific functions (shannon entropy, etc.) provide diversity functions that people are more familiar with. The general function is slightly different (as described in this project sheet and discussed in #341 for instance), but can be used to create the specific functions with a bit more work. The general function is fundamentally a better way of measuring diversity (we believe), but for historical reasons people still often use the specific ones.

**Reply by BakariMtili** [reply]
Thank you very much Richard, well understood
